src/DynamicMvvm.Abstractions/Command/IDynamicCommand.md

```diff
@@ -147,59 +147,62 @@ public static class MyCustomCommandDecoratorExtensions
+You can configure your commands locally using the various extension methods on `IDynamicCommandBuilder`.
 
 ```csharp
-// This will create a new IDynamicCommand using a IDynamicCommandFactory
-var myDynamicCommandFactory = new DynamicCommandFactory();
-var myCommandDecorator = new DynamicCommandStrategyDecorator(c => c.WithLogs());
-var myCommand = myDynamicCommandFactory.CreateFromAction("MyCommand", ExecuteCommand, myCommandDecorator);
+// This will create a new IDynamicCommand using a IDynamicCommandBuilderFactory and IDynamicCommandBuilder.
+var myFactory = new DynamicCommandBuilderFactory();
+var myBuilder = myFactory.CreateFromAction("MyCommand", ExecuteCommand)
+	.WithLogs();
+var myCommand = myBuilder.Build();
 private void ExecuteCommand() { }
 ```
 
-You can decorate all commands created from the `IDynamicCommandFactory` by using a **global** decorator.
+You can configure all commands created from the `IDynamicCommandBuilderFactory` by using a **default configuration**.
 
 ```csharp
-// This will create a new IDynamicCommand using a IDynamicCommandFactory
-var myCommandDecorator = new DynamicCommandStrategyDecorator(c => c
-  // This will add logs your commands.
-  .WithLogs()
+// This will create a new IDynamicCommand using a IDynamicCommandBuilderFactory and IDynamicCommandBuilder.
+var myFactory = new DynamicCommandBuilderFactory(c => c
+  // This will run your commands on a background thread.
+  .ExecuteOnBackgroundThread()
 
   // This will catch any errors unhandled by your commands.
   .CatchErrors(HandleCommandError)
 
-  // This will run your commands on a background thread.
-  .ExecuteOnBackgroundThread()
+  // This will add logs your commands.
+  .WithLogs()
 ));
 
-var myDynamicCommandFactory = new DynamicCommandFactory(myCommandDecorator);
-var myCommand = myDynamicCommandFactory.CreateFromAction("MyCommand", ExecuteCommand);
+var myBuilder = myFactory.CreateFromAction("MyCommand", ExecuteCommand);
+var myCommand = myBuilder.Build();
+
 private void ExecuteCommand() { }
 ```
 
-_Note: The **global** decorator wraps the **local** decorator which wraps the **implementation**. The hierarchy could look like the following._
+_Note: The **default** configuration wraps the **local** configuration which wraps the **implementation**. The hierarchy could look like the following._
 
 ```
 - MyCommand
-  - BackgroundCommandStrategy (global decorator)
-    - ErrorHandlerCommandStrategy (global decorator)
-      - LogCommandStrategy (global decorator)
-          - CanExecuteStrategy (local decorator)
-            - ... (local decorator)
-              - ActionCommandStrategy (implementation)
+  - BackgroundCommandStrategy (default config)
+    - ErrorHandlerCommandStrategy (default config)
+      - LogCommandStrategy (default config)
+        - CanExecuteStrategy (local config)
+          - ... (other local strategies)
+            - ActionCommandStrategy (implementation)
 ```
 
```

Comment:
Should we have an example on how to "remove" a decorator?

Reply:
https://github.com/nventive/Chinook.DynamicMvvm/pull/13